analytics_data_api/v0/documents.py

```diff
@@ -11,168 +11,60 @@ class RosterUpdate(Document):
-        settings = {
-            'number_of_shards': 1,
-            'number_of_replicas': 0
-        }
 
     @classmethod
     def get_last_updated(cls):
-        temp = cls.search().query('match', target_index=settings.ELASTICSEARCH_LEARNERS_INDEX).execute()
-        return temp
+        return cls.search().query('term', target_index=settings.ELASTICSEARCH_LEARNERS_INDEX).execute()
 
 
 class RosterEntry(Document):
     """
     Index which store learner information of a course.
     """
 
-    course_id = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    user_id = Integer(
-        fields={
-            'raw': Integer(),
-        }
-    )
-    username = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    name = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    email = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    language = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    location = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    year_of_birth = Integer(
-        fields={
-            'raw': Integer(),
-        }
-    )
-    level_of_education = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    gender = Text(
-        fields={
-            'raw': Text(),
-        }
-    )
-    mailing_address = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    city = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    country = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    goals = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    enrollment_mode = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    cohort = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )
-    segments = Keyword(
-        fields={
-            'raw': Keyword(),
-        }
-    )  # segments is an array/list of strings
-    problems_attempted = Integer(
-        fields={
-            'raw': Integer(),
-        }
-    )
-    problems_completed = Integer(
-        fields={
-            'raw': Integer(),
-        }
-    )
-    problem_attempts_per_completed = Float(
-        fields={
-            'raw': Float(),
-        }
-    )
+    course_id = Keyword()
+    user_id = Integer()
+    username = Keyword()
+    name = Keyword()
+    email = Keyword()
+    language = Keyword()
+    location = Keyword()
+    year_of_birth = Integer()
+    level_of_education = Keyword()
+    gender = Keyword()
+    mailing_address = Keyword()
+    city = Keyword()
+    country = Keyword()
+    goals = Keyword()
+    enrollment_mode = Keyword()
+    cohort = Keyword()
+    segments = Keyword()  # segments is an array/list of strings
+    problems_attempted = Integer()
```

Comment:
Please be aware that there is `short` integer type. So ofter `short` is more than enough to store your values. https://www.elastic.co/guide/en/elasticsearch/reference/current/number.html

Reply:
And one more. Observing you mapping I've seen the field 
```
"videos_viewed": {
                    "type": "long"
                },
```
1. I did not find the field in your described documents. Do I seek badly?
2. Why `long` type u are going to use?

Reply:
Thanks for remarks about mapping. In previous mapping scheme fields had `Integer` type, I don't think that we should change this type to another. Short type capacity is not enough for saving some learner values, for example `user_id` can be higher than 32